index.js

```diff
@@ -52,79 +52,79 @@ const root =
   // clicked on (in terminals such as iTerm) to open the file in an editor.
   `${process.cwd()}/`
 
-expandThunks = (array) => array.map(
-  elem => typeof elem == 'function' ? elem() : elem
-)
+const PRETTY = Boolean(process.env.PRETTY)
 
-makeLazyLogger = (strictLogger) => {
-  return strictLogger.enabled ?
-    // if logger enabled, call it with functions expanded to actual values
-    (...args) => strictLogger.apply(strictLogger, expandThunks(args)) :
-    // no-op
-    () => {}
-}
+const DEBUG = process.env.DEBUG
 
-makeStrictLogger = (label, {logToStderr, isErrorLogger}) => {
-  const logger = debug(label)
-  if (!logToStderr && !process.env.DEBUG_LOG_TO_STDERR) {
-    // by default log to stdout
-    logger.log = console.log.bind(console);
-  }
+const LEVELS = ['DEBUG', 'LOG', 'WARN', 'ERROR', 'EMERGENCY']
 
-  if (isErrorLogger) {
-    const errorLogger = (...args) => {
-      if (!logger.enabled) return
+const expandThunks = array => array.map(elem => (typeof elem == 'function' ? elem() : elem))
 
-      const last = args[ args.length - 1 ]
+const makeLazyLogger = strictLogger => {
+  // call it with functions expanded to actual values
+  return (...args) => strictLogger.apply(strictLogger, expandThunks(args))
+}
 
-      if (last instanceof Error) {
-        const argsWithErrorMessage = args.slice(0, -1)
-        argsWithErrorMessage.push(last.toString())
-        // Log error.message via the logger with label.
-        logger.apply(logger, argsWithErrorMessage)
-        // Log error object on its own, so that it can be parsed by GCP Error
-        // reporting.
-        console.error(last)
-      }
-      else {
-        logger(args)
+const parseArgs = args => {
+  const parsed = args.map(arg => {
+    const argument =
+      arg instanceof Error
+        ? {
+            name: arg.name,
+            message: arg.message,
+            data: arg.data,
+            stack: arg.stack
+          }
+        : arg
+    return PRETTY ? argument : stringify(argument)
+  })
+
+  return PRETTY ? parsed : parsed.join(' ')
+}
+
+const makeStrictLogger = (severity, context) => {
+  const logger = (...args) => {
+    const message = parseArgs(args)
+    const payload = { severity, timestamp: Date.now(), context, message }
+
+    if (
+      (!DEBUG && severity !== 'DEBUG') ||
+      (DEBUG && !LEVELS.includes(DEBUG) && severity !== 'DEBUG') ||
+      DEBUG === '*' ||
+      (LEVELS.includes(DEBUG) && DEBUG === severity)
+    ) {
```

Comment:
llthis is not efficient: this loggic is executed every single time the logging function is called, even though eventually we might end up not logging anything. Original logger implementation decides if particular logger instance is `enabled` or not at the instantiation time (once per process run) and then simply checks `.enabled` prop to decide if logging logic should be executed or not